File: Chatmvp/app/src/main/java/nosql/anew/chatmvp/models/MessageModel.java
package nosql.anew.chatmvp.models;

import com.google.firebase.database.DataSnapshot;

import java.util.ArrayList;

import nosql.anew.chatmvp.interfaces.ModelCallBacks;

/**
 * Created by saksham on 26/6/17.
 */

public class MessageModel {
    private ArrayList<ChatPojo> mMessages;

    public void addMessages(DataSnapshot dataSnapshot, ModelCallBacks callBacks){
        if (mMessages==null){
            mMessages= new ArrayList<>();
        }
        ChatPojo chatPojo=new ChatPojo(dataSnapshot);
        mMessages.add(chatPojo);
        callBacks.onModelUpdated(mMessages);
    }
}

File: Chat-Kotlinmvc/app/src/main/java/nosql/anew/kotlinmvc/models/MessageModel.kt
package nosql.anew.kotlinmvc.models

import com.google.firebase.database.DataSnapshot
import nosql.anew.kotlinmvc.interfaces.ModelCallBacks
import java.util.ArrayList

/**
 * Created by Sam on 11/04/18.
 */
class MessageModel(internal var mModelCallBacks: ModelCallBacks) {
    private var mMessages: ArrayList<ChatPojo>? = null

    fun addMessages(dataSnapshot: DataSnapshot) {
        if (mMessages == null) {
            mMessages = ArrayList()
        }

        val chatPojo = ChatPojo(dataSnapshot)
        mMessages!!.add(chatPojo)
        mModelCallBacks.onModelUpdated(mMessages!!)
    }
}